clouddriver-google/src/test/groovy/com/netflix/spinnaker/clouddriver/google/deploy/ops/GoogleUserDataProviderSpec.groovy

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2017 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.google.deploy.ops
+
+import com.google.api.services.compute.Compute
+import com.netflix.spinnaker.clouddriver.google.deploy.description.BasicGoogleDeployDescription
+import com.netflix.spinnaker.clouddriver.google.security.GoogleNamedAccountCredentials
+import spock.lang.Specification
+import spock.lang.Unroll
+
+@Unroll
+class GoogleUserDataProviderSpec extends Specification {
+
+  def "combines common and custom user data"() {
+    given:
+    def computeMock = Mock(Compute)
+
+    def credentials = new GoogleNamedAccountCredentials.Builder().accountType('test').environment('env').userDataFile("file").compute(computeMock).build()
+    def provider = Spy(GoogleUserDataProvider) {
+      it.getFileContents(_) >> commonUserData
+    }
+    def serverGroupName = 'app-stack-detail-v001'
+    def instanceTemplateName = 'app-stack-detail-v001-123'
+    def description = new BasicGoogleDeployDescription(region: 'region', accountName: 'account', credentials: credentials)
+
+    when:
+    def userData = provider.getUserData(serverGroupName, instanceTemplateName, description, credentials, customUserData)
+
+    then:
+    userData == expectedUserData
+    noExceptionThrown()
+
+    where:
+    customUserData            | commonUserData           || expectedUserData
+    null                      | null                      | [:]
+    ''                        | null                      | [:]
+    null                      | ''                        | [:]
+    null                      | 'user_data=common'        | ['user_data':'common']
+    ''                        | 'user_data=common'        | ['user_data':'common']
+    'user_data=common'        | null                      | ['user_data':'common']
+    'user_data=common'        | ''                        | ['user_data':'common']
+    'user_data=common'        | 'user_data=common'        | ['user_data':'common']
+    '%%account%%'             | '%%region%%'              | ['%%account%%':'%%account%%', 'region':'region']
```

Comment:
Oh boy. I don't recall why I made that decision. I am looking it over again to see if I can remember.